src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -178,7 +201,7 @@ func isCsrApproved(csr *certificatesv1beta1.CertificateSigningRequest) bool {
 	return false
 }
 
-func (c controller) PostInstallConfigs(wg *sync.WaitGroup) {
+func (c controller) PostInstallConfigs(wg *sync.WaitGroup, status *ControllerStatus) {
```

Comment:
I think we miss logic here. If we have cluster in error there is not point to continue to wait for finalizing. We can just exit it

Reply:
This edge case (of all hosts in error) usually occur when we timeout for waiting for control plane. I believe this exceeds all other timeouts?